cmd/loadTests.go

```diff
@@ -1069,66 +1068,139 @@ func (h *ConcreteHandlerResources) handleApplicationCreation(ctx *JourneyContext
 		ApplicationCreationTimeMaxPerThread[ctx.ThreadIndex] = applicationCreationTime
 	}
 
-	// Wait for
-	// 1. our application was created (condition.Type == "Created" && condition.Status == "True")
-	// or
-	// 2. our application to be in error (condition.Status == "True" && (strings.HasPrefix(condition.Type, "Error") || strings.HasSuffix(condition.Type, "Error"))
+	return h.validateApplicationCreation(ctx, framework, ApplicationName, username, usernamespace, applicationCreationTime)
+}
+
+func isConditionError(condition metav1.Condition) bool {
+	return condition.Status == "True" && (strings.HasPrefix(condition.Type, "Error") || strings.HasSuffix(condition.Type, "Error"))
+}
+
+// Define Structs for Grouping Parameters for "handleCondition" function to reduce number of parameters used
+type ConditionDetails struct {
+	Type   string
+	Status metav1.ConditionStatus
+}
+
+type CreationDetails struct {
+	Timestamp time.Time
+	Duration  time.Duration
+}
+
+type SuccessHandler interface {
+	HandleSuccess(ctx *JourneyContext, name string, timeInSeconds float64)
+}
+
+type ApplicationSuccessHandler struct{}
+
+func (h ApplicationSuccessHandler) HandleSuccess(ctx *JourneyContext, appName string, timeInSeconds float64) {
+	handleApplicationSuccess(ctx, appName, timeInSeconds)
+}
+
+type ItsSuccessHandler struct {
+	Username string
+}
+
+func (h ItsSuccessHandler) HandleSuccess(ctx *JourneyContext, itsName string, timeInSeconds float64) {
+	handleItsSuccess(ctx, itsName, h.Username, timeInSeconds)
+}
+
+type CdqSuccessHandler struct{}
+
+func (h CdqSuccessHandler) HandleSuccess(ctx *JourneyContext, cdqName string, timeInSeconds float64) {
+	handleCdqSuccess(ctx, cdqName, timeInSeconds)
+}
+
+type ComponentSuccessHandler struct {
+	Username string
+}
+
+func (h ComponentSuccessHandler) HandleSuccess(ctx *JourneyContext, componentName string, timeInSeconds float64) {
+	handleComponentSuccess(ctx, h.Username, componentName, timeInSeconds)
+}
+
+func handleCondition(condition metav1.Condition, ctx *JourneyContext, name string, creationDetails CreationDetails, conditionDetails ConditionDetails, successHandler SuccessHandler) (bool, error) {
+	if condition.Type == conditionDetails.Type && condition.Status == conditionDetails.Status {
+		actualCreationTimeInSeconds := CalculateActualCreationTimeInSeconds(condition.LastTransitionTime.Time, creationDetails.Timestamp, creationDetails.Duration)
+		successHandler.HandleSuccess(ctx, name, actualCreationTimeInSeconds)
+		return true, nil
+	}
+
+	if isConditionError(condition) {
+		return true, fmt.Errorf("%s %s is in Error state: %s", name, condition.Message)
```

Comment:
```suggestion
                return true, fmt.Errorf("%s is in Error state: %s", name, condition.Message)
```

I believe this will fix the golangci-lint error.